sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RowsResource.java

```diff
@@ -117,27 +181,96 @@
     }
     paramsB = paramsB.setPageSize(Int32Value.of(pageSize));
 
-    QueryOuterClass.Query query =
+    final QueryOuterClass.Query query =
         QueryOuterClass.Query.newBuilder().setParameters(paramsB.build()).setCql(cql).build();
-    QueryOuterClass.Response grpcResponse = blockingStub.executeQuery(query);
-
-    final QueryOuterClass.ResultSet rs = grpcResponse.getResultSet();
-    final int count = rs.getRowsCount();
-
-    String pageStateStr = null;
-    BytesValue pagingStateOut = rs.getPagingState();
-    if (pagingStateOut.isInitialized()) {
-      ByteString rawPS = pagingStateOut.getValue();
-      if (!rawPS.isEmpty()) {
-        byte[] b = rawPS.toByteArray();
-        pageStateStr = Base64.getEncoder().encodeToString(b);
-      }
+    return Sgv2RequestHandler.handle(
+        () -> {
+          QueryOuterClass.Response grpcResponse = blockingStub.executeQuery(query);
+
+          final QueryOuterClass.ResultSet rs = grpcResponse.getResultSet();
+          final int count = rs.getRowsCount();
+
+          String pageStateStr = null;
+          BytesValue pagingStateOut = rs.getPagingState();
+          if (pagingStateOut.isInitialized()) {
+            ByteString rawPS = pagingStateOut.getValue();
+            if (!rawPS.isEmpty()) {
+              byte[] b = rawPS.toByteArray();
+              pageStateStr = Base64.getEncoder().encodeToString(b);
+            }
+          }
```

Comment:
should we create a utility for this, could be used in lot of places? Actually we should have this already in the old code for sure?

Reply:
Agreed needs to be extracted. The challenge with old code tends to be dependencies, i.e. where exactly it is used; I can see how easily it could be reused.

Reply:
just for reference I was pointing to this -> `io.stargate.core.util.ByteBufferUtils#toBase64`

Reply:
I'm thinking of maybe adding `byte[]` taking argument; and should really optimize this:

```
    if (base64.chars().anyMatch(c -> c == '/' || c == '+' || c == ' ')) {
```

since that is painfully slow way to check anything (I investigated it a bit earlier https://cowtowncoder.medium.com/measuring-string-indexofany-string-performance-java-fecb9eb473fa -- could use "commons-lang" variant or switch.

Reply:
Filed #1404.

Reply:
where do we have this actually?

Reply:
`ByteBufferUtils`